source/agora/consensus/protocol/Nominator.d

```diff
@@ -521,3 +521,271 @@ extern(D):
         });
     }
 }
+
+/// Adds hashing support to SCPStatement
+private struct SCPStatementHash
+{
+    // sanity check in case a new field gets added.
+    // todo: use .tupleof tricks for a more reliable field layout change check
+    static assert(SCPNomination.sizeof == 80);
+
+    /// instance pointer
+    private const SCPStatement* st;
+
+    /// Ctor
+    public this (const SCPStatement* st) @nogc nothrow @trusted
```

Comment:
```suggestion
    public this (const SCPStatement* st) @nogc nothrow @safe pure
```

No ?

Reply:
I did not really know ctors can be pure. I'm also surprised a function can be safe when assigning pointers. Oh well.